### PR TITLE
update secp256k1 (31994c8)

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

**Comment by fanatid** [comment]
hm, some strange things with 0.11 on travis,
on my laptop:

``` bash
$ nvm use 0.11
Now using node v0.11.16 (npm v2.3.0)
$ node --version
v0.11.16
$ npm --version
2.3.0
$ rm -rf node_modules/ build/
$ npm i
...
$ npm run test

> secp256k1@1.1.5 test /home/kirill/coloredcoins/secp256k1-node
> standard ./index.js && ./node_modules/mocha/bin/mocha --reporter spec ./test/



  it should handle basic ecdsa ops
    ✓ should create a public key (82ms)
    ✓ should sign a message with a DER sig (75ms)
    ✓ should sign a message async (73ms)
    ✓ should verify a signature (45ms)
    ✓ should NOT verify an invalid signature (49ms)
    ✓ should verify a signature async (45ms)
    ✓ should create a compact signature (191ms)
    ✓ should create a compact signature async (182ms)
    ✓ should recover a compact signature and return the public key
    ✓ should recover a compact signature and return the public key, async

  invalid inputs
    ✓ should not crash when recoverId is out of bounds - sync
    ✓ should not crash when recoverId is out of bounds - async
    ✓ should not crash when giving it an undefined private key
    ✓ should not crash when given an undefined privateKey in createPublicKey
    ✓ should not crash when given an undefined publickeys in verify
    ✓ should not crash when sign tx with undefined


  16 passing (761ms)
```

**Comment by wanderer** [comment]
cool thanks @fanatid ! What is the point of changing the `compressed` Bool to a `flags` int?

File: functions.cc [hunk 1]
```diff
@@ -84,35 +86,35 @@ NAN_METHOD(Sign){
 
 NAN_METHOD(Recover){
   Nan::HandleScope scope;
-  
+
   const unsigned char *msg = (unsigned char *) node::Buffer::Data(info[0].As<Object>());
   Local<Object> sig_buf = info[1].As<Object>();
   int recid = info[2]->IntegerValue();
-  bool compressed = info[3]->IntegerValue();
+  unsigned int flags = info[1]->BooleanValue() ? SECP256K1_EC_COMPRESSED : 0;
```
Comment (wanderer) [inline functions.cc]: why `info[1]` ? isn't info[1] the signature?

**Comment by fanatid** [comment]
It was copy-paste, fixed this. :blush:
Now travis fails with all node versions. I tried run tests locally a few times and also got error, I will try to fix this.

**Comment by fanatid** [comment]
ok, now code pass all tests for all versions, but as I see new [secp256k1](https://github.com/bitcoin/secp256k1/) hasn't functional for recovering public key from `DER` signature.. so I added [throwing error](https://github.com/fanatid/secp256k1-node/blob/bacca0e1b55daa3f5da76a0ae7d78f4faf225ce0/functions.cc#L97)

**Comment by wanderer** [comment]
Looks good, Merged! :+1:

**Comment by wanderer** [comment]
ok published as 1.1.6

**Pull request merged**